src/tables/math.rs

```diff
@@ -0,0 +1,1062 @@
+//! A [MATH Table](https://docs.microsoft.com/en-us/typography/opentype/spec/math) implementation.
+
+use crate::GlyphId;
+use crate::gpos::Device;
+use crate::opentype_layout::Coverage;
+use crate::parser::{
+    Offset16, Stream,
+    FromSlice, FromData, Offset,
+    LazyArray16, LazyOffsetArray16,
+};
+
+/// A [MATH Table](https://docs.microsoft.com/en-us/typography/opentype/spec/math).
+#[allow(missing_debug_implementations)]
+#[allow(missing_docs)]
+#[derive(Clone)]
+pub struct Table<'a> {
+    pub major_version: u8,
+    pub minor_version: u8,
+    pub math_constants: MathConstants<'a>,
+    pub math_glyph_info: MathGlyphInfo<'a>,
+    pub math_variants: MathVariants<'a>,
+}
+
+#[derive(Debug, Default, Copy, Clone)]
+struct RawMathValueRecord {
+    value: i16,
+    device_table_offset: Option<Offset16>,
+}
+
+impl FromData for RawMathValueRecord {
+    const SIZE: usize = 4;
+    fn parse(data: &[u8]) -> Option<Self> {
+        let mut s = Stream::new(data);
+        Some(RawMathValueRecord {
+            value: s.read()?,
```

Comment:
Please use explicit types like `s.read::<i16>()?`

Reply:
I have made all `Stream::{read, parse_at_offset16}` calls explicit about the target types.